### ISSUE TITLE
Zip build

### ISSUE THREAD

**Issue by 89z** [malxau/yori#103]
Do you zip build available? Similar to these:

- https://github.com/nushell/nushell/releases/tag/0.44.0
- https://github.com/PowerShell/PowerShell/releases/tag/v7.2.1

**Comment by malxau** [comment]
There's some discussion about this in #6.

Yori uses `.cab` format for packages, so once a package is downloaded, extracting executables is straightforward.  These packages are not hosted on Github - I really want to a) maintain a single source for downloads and b) not have it be a service which could vanish tomorrow.  Within the constraint of having a single source, I'm totally open to suggestions about how to support the broadest set of use cases.

Today, `ypm -download-stable` is probably the "easiest" way to download these packages, and from there, contents can be extracted and copied wherever.

Part of the conversation in #6, which still confuses me a little, is the idea that somebody wants to download binaries but doesn't trust executing the binaries to download the binaries.  But if the binaries aren't trusted for execution, what's the point of downloading them? Ultimately the solution to that is building from source, because the issue is really about trusting binaries, not their delivery mechanism.

**Comment by 89z** [comment]
> Part of the conversation in #6, which still confuses me a little, is the idea that somebody wants to download binaries but doesn't trust executing the binaries to download the binaries. But if the binaries aren't trusted for execution, what's the point of downloading them?

Taking Yori as an example, lets say hypothetically that you did offer a Zip file, and inside that Zip file was just `yori.exe`. When the user runs that file, it runs the shell.

In that situation, the assumption for anyone downloading the Zip file, is that the executable is just going to be a shell, nothing else. So the scope of the executable is well defined.

Contrast that with the current sitation, where users are forced to download an run an installer. An installer is quite different from the previous example, as an installer can take basically arbitrary action on the host system. It can write registry entries, create folders, create files, DELETE files, or any number of other actions. In the previous example, technically the `yori.exe` could do the same thing, but the expectation is that the program is just going to act as a shell.

Contrast with an installer, where the expectation is that the installer can and will do any number of weird things to your system. Finally, with a Zip file, if you dont like the containing program, then you just delete the Zip file, delete the contents, and your done. With an installer, who knows what the installer did, so undoing an installers actions can be difficult, especially is no uninstaller is provided.

So in summary, a Zip file is a "low trust" way for a user to try a new program. User is still taking a risk of course, but the expectation is that the zipped executable is staying in its advertised domain.

**Comment by 89z** [comment]
I think the better question is, why do most programs (like the ones I linked in the original comment) offer a Zip, and Yori doesnt? Or even just look at Linux, where archives are the default (`tar.gz` or similar).

**Comment by malxau** [comment]
There's a few things here.

First, I agree that there appears to be a big difference in expectations about which executables do what, although as you note, this isn't so much a technical thing as an expectation thing.  As I wrote in the last issue, the installer here will not modify the registry or perform any system level change unless it is explicitly requested to do so.  One of those actions is to create an uninstall entry.  Nobody is forced to use the installer - it's just an easier frontend to download 50 packages than doing it manually.

As far as the better question though, it might be easier to distinguish what happens on Linux and what happens on Windows.

On Linux, a software developer typically releases a single source tarball.  Some (very large) projects can go beyond that, but an individual developer rarely would.  Distributions are free to compile that code and package it however they wish.  From my point of view, this is a good system: software developers release things once.

On Windows, users are typically not going to compile code, so developers are expected to provide both source and binaries.  There are many package managers out there, and individual users often have a preference, but there is typically not a distribution community that repackages things into desired formats.  So, software developers will be asked to package for each type of user - some might want a zip, others want Chocolatey, Scoop, WinGet, MS Store, you name it.  The most frustrating part is that since many users don't want a package manager, applications end up building their own update mechanism, and that mechanism conflicts with the rules or expectations around each packaging format.  If a user wanted a zip to avoid an installer for example, the last thing they'd want is an updater, but the binaries would ordinarily include it unless they're altered to not include it.

The higher level point is a volunteer run project depends on what people are motivated to do.  I am not going to make it hard for anyone to distribute this code however they'd like.  That does not mean I'm volunteering to distribute it in a multitude of ways.  Realistically, the more burden each release becomes, the fewer releases will occur.  I'd far sooner be spending energy on the next piece of functionality than spending energy trying to repackage the last piece of functionality.  If other people want to volunteer to contribute with a different focus, they are free to do so.
